src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useFormContext } from 'react-hook-form';
+import { Box } from '@mui/material';
+
+import { Svg, Table, UiType } from 'shared/components';
+import {
+  StyledBodyMedium,
+  StyledSvgPrimaryColorBtn,
+  StyledTitleMedium,
+  theme,
+  variables,
+} from 'shared/styles';
+import { useCurrentActivity } from 'modules/Builder/hooks';
+import { FlankerStimulusSettings } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.types';
+import { exportTemplate } from 'shared/utils';
+
+import { RoundType } from '../../RoundSettings.types';
+import {
+  getRoundBlocks,
+  getSequencesData,
+  getSequencesHeadCells,
+  getUploadedTableRows,
+} from './BlockSequencesContent.utils';
+import { ImportSequencesPopup } from './ImportSequencesPopup';
+import { ImportSequencesType } from './ImportSequencesPopup/ImportSequencesPopup.types';
+import { UploadedTable } from './BlockSequencesContent.types';
+
+export const BlockSequencesContent = ({ isPracticeRound }: RoundType) => {
+  const { t } = useTranslation();
+  const [importTableVisible, setImportTableVisible] = useState(false);
+  const [uploadedTable, setUploadedTable] = useState<UploadedTable>(null);
+  const { watch, setValue } = useFormContext();
+  const { fieldName } = useCurrentActivity();
+  const stimulusField = `${fieldName}.general.stimulusTrials`;
+  const stimulusTrials: FlankerStimulusSettings[] = watch(stimulusField);
+  const blockSequencesField = `${fieldName}.${isPracticeRound ? 'practice' : 'test'}.blocks`;
+
+  const { defaultExportTable, defaultTableRows } = getSequencesData(stimulusTrials);
+  const tableRows = uploadedTable ? getUploadedTableRows(uploadedTable) : defaultTableRows || [];
+
+  const tableWrapperStyles = { opacity: uploadedTable ? '1' : '0.5' };
+  const importSequencesUiType = uploadedTable
+    ? ImportSequencesType.Update
+    : ImportSequencesType.Upload;
+  const btnIconId = uploadedTable ? 'edit' : 'add';
+  const btnText = t(`${uploadedTable ? 'update' : 'upload'}`);
+
+  useEffect(() => {
+    if (!uploadedTable?.length) return;
+    const blocks = getRoundBlocks(stimulusTrials, uploadedTable);
+    if (!blocks) return;
+
+    setValue(blockSequencesField, blocks);
+  }, [uploadedTable, stimulusTrials]);
+
+  return stimulusTrials?.some((trial) => !!trial.imageName) ? (
+    <>
+      <Box>
```

Comment:
Do y need the Box?

Reply:
yes for btn to not occupy the full width of the section